backend/src/services/resourceService.js

```diff
@@ -105,5 +67,43 @@ export const resourceService = {
     } else {
       throw { error: 'Kingdom ID is required.' };
     }
-  },
 };
+
+const updateResourceByType = async (kingdomId, cost, genAdd, {type, amount, generation, updatedAt}) => {
+  const updateAmount = (cost !== null) ? amount - cost : calculateNewAmount(amount, generation, updatedAt);
+  const updateGeneration = genAdd ? generation + genAdd : generation;
+  const changedRows = (await resourceRepo.updateResource({amount: updateAmount, generation: updateGeneration, kingdom_id : kingdomId, type : type}) ).changedRows;
+  if (changedRows === 0) throw new Error('Data not found.');
+};
+
+const typeParams = (kingdomId, cost, genAdd, resources) => ({
+  '-': [
+    [ kingdomId, cost, 0, resources.filter(e => e.type === 'gold')[0] ]
+  ],
+  'gold': [
+    [ kingdomId, cost, genAdd, resources.filter(e => e.type === 'gold')[0] ]
+  ],
+  'food': [
+    [ kingdomId, cost, 0, resources.filter(e => e.type === 'gold')[0] ],
+    [ kingdomId, 0, genAdd, resources.filter(e => e.type === 'food')[0] ]
+  ],
+  'default': [
+    [ kingdomId, null, 0, resources.filter(e => e.type === 'gold')[0] ],
+    [ kingdomId, null, 0, resources.filter(e => e.type === 'food')[0]  ]
+  ]
+});
```

Comment:
@gulyaasjaanos this looks like some weird magic, what's the point?
also `resources.filter(...)[0]` is `resources.find(...)`

Reply:
@gulyaasjaanos 
- generateResources
- spendGold
- updateGoldGeneration
- spendFood
- updateFoodGeneration

// buildingService - add Mine
resourcesService.spendGold(10)
resourcesService.updateFoodGeneration(1)